tests/test_uri.py

```diff
@@ -63,6 +64,10 @@ def testFile(self):
         self.assertFalse(uri.exists(), f"{uri} should not exist")
         self.assertEqual(uri.ospath, file)
 
+        path = pathlib.Path(file)
+        uri = ButlerURI(path)
+        self.assertEqual(uri.ospath, str(path))
```

Comment:
Not sure what the creation of the `BtlerURI`, but just wanted to check that this is still OK with/without trailing slashes. If anything the test is more fragile, so it's clearly fine, but just thought I'd flag that potential difference, depending on the value of path in future.

Reply:
ButlerURI is what is used as soon as anything arrives inside butler. It's the class that handles the abstraction of local file vs remote object store.

Reply:
I'm not sure I understand what you mean about fragility. I do see that I should be comparing `uri.ospath` to `file` so I will fix that.

Reply:
It's fine, I just didn't know exactly how equality was being done internally, so wanted to make sure this was trailing-slash safe (though I'd have been surprised if it wasn't). All good.